examples/utils.cpp

```diff
@@ -328,3 +328,113 @@ gpt_vocab::id gpt_sample_top_k_top_p(
 
     return logits_id[idx].second;
 }
+
+size_t ggml_quantize_q4_0(float * src, void * dst, int n, int k, int qk, int64_t * hist) {
```

Comment:
Hm, can you explain more about this method?

Reply:
Specifically, I don't really understand how the block size is ensured to be compatible in general - qk is 32 in the examples, but how is that? Also, if k is the size of the first dimension of the tensor, why are all the weights' tensors a multiple of 32? Sorry to be a bother but I'm looking at using ggml for a project and I'm not sure I understand this part.